runtime/compiler/control/HookedByTheJit.cpp

```diff
@@ -5922,6 +5922,13 @@ static int32_t J9THREAD_PROC samplerThreadProc(void * entryarg)
             javacoreData->numAotDataEntries,
             javacoreData->numJitHints,
             javacoreData->numJitProfiles);
+#if defined(OMR_GC_COMPRESSED_POINTERS)
+         TR_AOTHeader *hdrInCache = compInfo->getCompInfoForThread(samplerThread)->reloRuntime()->getStoredAOTHeader(samplerThread);
```

Comment:
This seems like a roundabout way of getting to the cached header, but I couldn't see another way

Reply:
Does this work? my expectation is that `compInfo->getCompInfoForThread(samplerThread)` will return NULL.

Reply:
You need an object of type `TR_SharedCacheRelocationRuntime`. I see that there is one in `CompilationInfo`. 
It can be obtained with `TR_SharedCacheRelocationRuntime  *reloRuntime() { return &_sharedCacheReloRuntime; }`. However, I don't see who is constructing it.

Reply:
Looking at the definition:
```
const TR_AOTHeader *
TR_SharedCacheRelocationRuntime::getStoredAOTHeader(J9VMThread *curThread)
   {
   J9SharedDataDescriptor firstDescriptor;
   firstDescriptor.address = NULL;
   javaVM()->sharedClassConfig->findSharedData(curThread, aotHeaderKey, aotHeaderKeyLength,
                                               J9SHR_DATA_TYPE_AOTHEADER, FALSE, &firstDescriptor, NULL);
   return (const TR_AOTHeader *)firstDescriptor.address;
   }
```
I don't see anything that depends on the TR_SharedCacheRelocationRuntime object.
You can create a static function that receives `sharedClassConfig` and `curThread` as parameters and returns the AOT header. Then change `TR_SharedCacheRelocationRuntime::getStoredAOTHeader(J9VMThread *curThread)` to use this new static function internally.